build-system/tasks/sweep-experiments/index.js

```diff
@@ -0,0 +1,480 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const argv = require('minimist')(process.argv.slice(2));
+const fastGlob = require('fast-glob');
+const log = require('fancy-log');
+const path = require('path');
+const {cyan, magenta, yellow} = require('ansi-colors');
+const {getOutput} = require('../../common/process');
+const {readJsonSync, writeFileSync} = require('fs-extra');
+
+const containRuntimeSource = ['3p', 'ads', 'extensions', 'src', 'test'];
+const containExampleHtml = ['examples', 'test'];
+
+const experimentsConfigPath = 'tools/experiments/experiments-config.js';
+const prodConfigPath = 'build-system/global-configs/prod-config.json';
+const canaryConfigPath = 'build-system/global-configs/canary-config.json';
+
+const globalWritablePaths = [
+  experimentsConfigPath,
+  prodConfigPath,
+  canaryConfigPath,
+];
+
+/**
+ * Ignores experiments that cannot be removed automatically.
+ * @param {string} id
+ * @return {boolean}
+ */
+const isSpecialCannotBeRemoved = (id) =>
+  // These are passed through to a third-party, so we can't determine whether
+  // they're still in use by looking at this repo alone.
+  // See https://git.io/JIBeB (amp-subscriptions-google.js)
+  id.startsWith('swg-');
```

Comment:
Unfortunately we need to special case these. @jpettitt

Reply:
Contrext: swg.js, included in amp-subscriptions-google looks for swg-* experiment flags. This allows those experiments through.  Since the experiments are in a 3p library they won't be visible to amp tooling.